services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +45,48 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  if(event == NULL) return ERR_CODE_INVALID_ARG;
+  if (thermalMgrQueueHandle == NULL) return ERR_CODE_INVALID_ARG;
```

Comment:
ERR_CODE_INVALID_STATE would be more appropriate.
```suggestion
  if (thermalMgrQueueHandle == NULL) return ERR_CODE_INVALID_STATE;
```